app/src/main/java/com/endava/flights/MainFlightActivity.kt

```diff
@@ -16,7 +28,13 @@ import androidx.fragment.app.FragmentActivity
  * Bonus track: Think one use case as a challenge for your partner.
  *
  */
-class MainFlightActivity : FragmentActivity() {
+class MainFlightActivity(private var viewModel : MainFlightViewModel = MainFlightViewModel()) : FragmentActivity() {
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
 
+        viewModel.viewModelScope.launch {
```

Comment:
why do you use here (the activity/UI) the viewModelScope. Could we consider leave the responsibility to the view model? (about launch coroutines)